machine_learning_hep/analysis/analyzer_jet.py

```diff
@@ -358,23 +422,184 @@ def fit(self):
                     sgn_func.Draw("same")
                     bkg_func.Draw("same")
-                latex = TLatex(0.2, 0.83, "%g #leq %s < %g GeV/#it{c}" % (self.lvar2_binmin_reco[ibin2], self.p_latexbin2var, self.lvar2_binmax_reco[ibin2]))
-                draw_latex(latex)
-                latex2 = TLatex(0.5, 0.83, "%g #leq #it{p}_{T, %s} < %g GeV/#it{c}" % \
-                    (self.lpt_finbinmin[ipt], self.p_latexnhadron, min(self.lpt_finbinmax[ipt], self.lvar2_binmax_reco[ibin2])))
-                draw_latex(latex2)
+                if (not np.isnan(bkg) and not np.isnan(sig) and not np.isnan(s_to_b)):
+                    latex3 = TLatex(0.67, 0.78, "mean = %s, #sigma  = %s" % \
+                            (round(mean, 2), round(sigma, 2)))
+                    draw_latex(latex3)
+                    latex4 = TLatex(0.29, 0.83, "B(mean #pm 2#sigma)= %s, S(mean #pm 2#sigma)  = %s, S/#sqrt{S+B} = %s" % \
+                            (int(bkg), int(sig), round(s_to_b, 2)))
+                    draw_latex(latex4)
+                    latex = TLatex(0.71, 0.72, "%g #leq %s < %g GeV/#it{c}" % (self.lvar2_binmin_reco[ibin2], self.p_latexbin2var, self.lvar2_binmax_reco[ibin2]))
+                    draw_latex(latex)
+                    latex2 = TLatex(0.71, 0.67, "%g #leq #it{p}_{T, %s} < %g GeV/#it{c}" % \
+                        (self.lpt_finbinmin[ipt], self.p_latexnhadron, min(self.lpt_finbinmax[ipt], self.lvar2_binmax_reco[ibin2])))
+                    draw_latex(latex2)
                 c_fitted_result.SaveAs("%s/fit_%s.eps" % (self.d_resultsallpdata, suffix_plot))
         myfilemc.Close()
         myfile.Close()
         fileout.Close()
         gROOT.SetBatch(tmp_is_root_batch)
 
-    def efficiency(self): # pylint: disable=too-many-branches, too-many-locals
+    def efficiency(self):
+
+        self.loadstyle()
+        lfileeff = TFile.Open(self.n_fileeff)
+        if not lfileeff:
+            self.logger.fatal(make_message_notfound(self.n_fileeff))
+        fileouteff = TFile.Open(self.file_efficiency, "recreate")
+        if not fileouteff:
+            self.logger.fatal(make_message_notfound(self.file_efficiency))
+
+        #string_shaperange = "%g #leq %s < %g" % (self.lvarshape_binmin_reco[0], self.v_varshape_latex, self.lvarshape_binmax_reco[-1])
+
+        cEff = TCanvas("cEff", "The Fit Canvas")
+        setup_canvas(cEff)
+        legeff = TLegend(.13, .65, .5, .88)
+        setup_legend(legeff)
+        list_his = []
+        list_ptcand_gen_old = []
+        list_ptcand_rec_old = []
+        list_ptcand_eff_old = []
+        for ibin2 in range(self.p_nbin2_reco):
+            stringbin2 = "_%s_%.2f_%.2f" % (self.v_var2_binning, \
+                                            self.lvar2_binmin_reco[ibin2], \
+                                            self.lvar2_binmax_reco[ibin2])
+            h_gen_pr = lfileeff.Get("h_gen_pr" + stringbin2)
+            h_gen_pr_old = h_gen_pr.Clone(h_gen_pr.GetName() + "_old")
```

Comment:
make suffix more explicit (but keep short)

Reply:
changed to h_gen_pr_inc